xds/server.go

```diff
@@ -220,17 +221,20 @@ func (s *GRPCServer) newListenerWrapper(lis net.Listener) (*listenerWrapper, err
 	// or not.
 	goodUpdate := grpcsync.NewEvent()
 
-	// The resource_name in the LDS request sent by the xDS-enabled gRPC server
-	// is of the following format:
-	// "/path/to/resource?udpa.resource.listening_address=IP:Port". The
-	// `/path/to/resource` part of the name is sourced from the bootstrap config
-	// field `grpc_server_resource_name_id`. If this field is not specified in
-	// the bootstrap file, we will use a default of `grpc/server`.
-	path := "grpc/server"
-	if cfg := s.xdsC.BootstrapConfig(); cfg != nil && cfg.ServerResourceNameID != "" {
-		path = cfg.ServerResourceNameID
+	// The server listener resource name template from the bootstrap
+	// configuration contains a template for the name of the Listener resource
+	// to subscribe to for a gRPC server. If the token `%s` is present in the
+	// string, it will be replaced with the server's listening "IP:port" (e.g.,
+	// "0.0.0.0:8080", "[::]:8080"). The absence of a template will be treated
+	// as an error since we do not have any default value for this.
+	cfg := s.xdsC.BootstrapConfig()
+	if cfg == nil || cfg.ServerListenerResourceNameTemplate == "" {
+		return nil, errors.New("missing server_listener_resource_name_template in the bootstrap configuration")
+	}
+	name := cfg.ServerListenerResourceNameTemplate
+	if strings.Contains(cfg.ServerListenerResourceNameTemplate, "%s") {
+		name = strings.Replace(cfg.ServerListenerResourceNameTemplate, "%s", lis.Addr().String(), 1)
```

Comment:
This should replace all after https://github.com/grpc/proposal/pull/230, right?

And just blindly call `Replace`?

Reply:
Saw this go by this morning. But forgot to fix this. Thanks.